airbyte-integrations/connectors/destination-bigquery/src/test-integration/java/io/airbyte/integrations/destination/bigquery/typing_deduping/AbstractBigQueryTypingDedupingTest.java

```diff
@@ -154,6 +162,64 @@ public void testRemovingPKNonNullIndexes() throws Exception {
     assertEquals(1, dumpFinalTableRecords(getStreamNamespace(), getStreamName()).toArray().length);
   }
 
+  @Test
+  public void testAirbyteMetaAndGenerationIdMigration() throws Exception {
+    final ConfiguredAirbyteCatalog catalog = new ConfiguredAirbyteCatalog().withStreams(List.of(
+        new ConfiguredAirbyteStream()
+            .withSyncMode(SyncMode.FULL_REFRESH)
+            .withDestinationSyncMode(DestinationSyncMode.APPEND)
+            .withSyncId(42L)
+            .withGenerationId(43L)
+            .withMinimumGenerationId(0L)
+            .withStream(new AirbyteStream()
+                .withNamespace(getStreamNamespace())
+                .withName(getStreamName())
+                .withJsonSchema(SCHEMA))));
+
+    // First sync
+    final List<AirbyteMessage> messages1 = readMessages("dat/sync1_messages.jsonl");
+    runSync(catalog, messages1, "airbyte/destination-bigquery:2.4.20");
+
+    // Second sync
+    final List<AirbyteMessage> messages2 = readMessages("dat/sync2_messages.jsonl");
+    runSync(catalog, messages2);
+
+    // The first 5 records in these files were written by the old version, and have
```

Comment:
I could be convinced to make another copy of the expectedrecords files for this change, just didn't really feel like it :shrug: